File: baselibrary/src/main/java/com/aslan/baselibrary/http/error/HttpErrorFunctionObservable.java
package com.aslan.baselibrary.http.error;

import android.content.Context;
import com.aslan.baselibrary.base.DataError;
import io.reactivex.Observable;

/**
 * 处理http异常
 *
 * @author Aslan chenhengfei@yy.com
 * @date 2020/6/10
 */
public class HttpErrorFunctionObservable<T> extends BaseHttpErrorFunction<Observable<T>> {

  public HttpErrorFunctionObservable(Context context) {
    super(context);
  }

  @Override
  public Observable<T> error(DataError ex) {
    return Observable.error(ex);
  }
}

File: baselibrary/src/main/java/com/aslan/baselibrary/http/error/HttpErrorFunctionObservable.kt
package com.aslan.baselibrary.http.error

import android.content.Context
import io.reactivex.Observable

/**
 * 处理http异常
 *
 * @author Aslan chenhengfei@yy.com
 * @date 2020/6/10
 */
open class HttpErrorFunctionObservable<T>(context: Context) :
    BaseHttpErrorFunction<Observable<T>>(context) {
    override fun error(ex: Exception): Observable<T> {
        return Observable.error(ex)
    }
}